src/displayapp/screens/Weather.cpp

```diff
@@ -0,0 +1,203 @@
+        maxTemp = Controllers::SimpleWeatherService::CelsiusToFahrenheit(maxTemp);
+        tempUnit = 'F';
+      }
+      temp = temp / 100 + (temp % 100 >= 50 ? 1 : 0);
+      maxTemp = maxTemp / 100 + (maxTemp % 100 >= 50 ? 1 : 0);
+      minTemp = minTemp / 100 + (minTemp % 100 >= 50 ? 1 : 0);
+      lv_label_set_text(icon, Symbols::GetSymbol(optCurrentWeather->iconId));
+      lv_label_set_text(condition, Symbols::GetCondition(optCurrentWeather->iconId));
+      lv_label_set_text_fmt(temperature, "%d°%c", temp, tempUnit);
+      lv_label_set_text_fmt(minTemperature, "%d°", minTemp);
+      lv_label_set_text_fmt(maxTemperature, "%d°", maxTemp);
+    } else {
+      lv_label_set_text(icon, "");
+      lv_label_set_text(condition, "");
+      lv_label_set_text(temperature, "---");
+      lv_obj_set_style_local_text_color(temperature, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
+      lv_label_set_text(minTemperature, "");
+      lv_label_set_text(maxTemperature, "");
+    }
+  }
+
+  currentForecast = weatherService.GetForecast();
+  if (currentForecast.IsUpdated()) {
+    auto optCurrentForecast = currentForecast.Get();
+    if (optCurrentForecast) {
+      std::tm localTime = *std::localtime((const time_t*) &optCurrentForecast->timestamp);
+
+      for (int i = 0; i < Controllers::SimpleWeatherService::MaxNbForecastDays; i++) {
+        int16_t maxTemp = optCurrentForecast->days[i].maxTemperature;
+        int16_t minTemp = optCurrentForecast->days[i].minTemperature;
+        lv_table_set_cell_type(forecast, 2, i, temperatureStyle(maxTemp));
+        lv_table_set_cell_type(forecast, 3, i, temperatureStyle(minTemp));
+        if (settingsController.GetWeatherFormat() == Controllers::Settings::WeatherFormat::Imperial) {
+          maxTemp = Controllers::SimpleWeatherService::CelsiusToFahrenheit(maxTemp);
+          minTemp = Controllers::SimpleWeatherService::CelsiusToFahrenheit(minTemp);
+        }
+        uint8_t wday = localTime.tm_wday + i + 1;
+        if (wday > 6) {
+          wday -= 7;
+        }
+        maxTemp = maxTemp / 100 + (maxTemp % 100 >= 50 ? 1 : 0);
+        minTemp = minTemp / 100 + (minTemp % 100 >= 50 ? 1 : 0);
+        lv_table_set_cell_value(forecast, 0, i, WeekDays[wday]);
+        lv_table_set_cell_value(forecast, 1, i, Symbols::GetSymbol(optCurrentForecast->days[i].iconId));
+        // Pad cells based on the largest number of digits on each column
+        std::string maxPadding = "";
+        std::string minPadding = "";
+        int maxSize = snprintf(nullptr, 0, "%d", maxTemp);
+        int minSize = snprintf(nullptr, 0, "%d", minTemp);
+        switch (maxSize - minSize) {
+          case -2:
+            maxPadding = "  ";
+            break;
+          case -1:
+            maxPadding = " ";
+            break;
+          case 1:
+            minPadding = " ";
+            break;
+          case 2:
+            minPadding = "  ";
+            break;
+        }
```

Comment:
I don't really like using `std::string` here, as it can quite easily be done with a C array.

```suggestion
        // Pad cells based on the largest number of digits on each column
        char maxPadding[3] = "  ";
        char minPadding[3] = "  ";
        int maxSize = snprintf(nullptr, 0, "%d", maxTemp);
        int minSize = snprintf(nullptr, 0, "%d", minTemp);
        switch (maxSize - minSize) {
          case -1:
            maxPadding[1] = '\0';
            break;
          case 1:
            minPadding[1] = '\0';
            break;
        }
```

Reply:
This is missing
```cpp
  case 0:
    minPadding[0] = '\0';
    maxPadding[0] = '\0';
```

We also might want to handle longer numbers correctly, even though they are most likely wrong. This could be done by changing the `case 0` to `default`.

Reply:
Oh yeah! Forgot about that

Reply:
I've just updated it to handle larger numbers without having to add cases for them.

Reply:
I love this, thanks!